itests/kit/ensemble.go

```diff
@@ -1099,17 +1128,42 @@ func (n *Ensemble) BeginMirMiningWithDelayForFaultyNodes(ctx context.Context, wg
 
 		go func(ctx context.Context, i int, m *TestMiner) {
 			defer wg.Done()
-			m.mirMembership = membership
+			m.mirMembership = membershipString
 			m.mirDB = NewTestDB()
 			m.checkpointPeriod = len(miners) + len(faultyMiners)
 			cfg := mir.Config{
-				MembershipCfg:    mir.MembershipFromStr(membership),
-				CheckpointPeriod: m.checkpointPeriod,
+				SegmentLength: 1,
 			}
+			membership := validator.StringMembership(membershipString)
 			if i > len(miners) && delay > 0 {
 				RandomDelay(delay)
 			}
-			err := mir.Mine(ctx, m.mirAddr, m.mirHost, m.FullNode, m.mirDB, &cfg)
+			err := mir.Mine(ctx, m.mirAddr, m.mirHost, m.FullNode, m.mirDB, membership, &cfg)
+			if xerrors.Is(mapi.ErrStopped, err) {
+				return
+			}
+			require.NoError(n.t, err)
+		}(ctx, i, m)
+	}
+}
+
+func (n *Ensemble) BeginMirMiningWithMembershipFromFile(ctx context.Context, configFileName string, wg *sync.WaitGroup, checkpoint int, miners []*TestMiner, faultyMiners ...*TestMiner) {
+	for i, m := range append(miners, faultyMiners...) {
+		ctx, cancel := context.WithCancel(ctx)
+		m.stopMir = cancel
+
+		wg.Add(1)
+
+		go func(ctx context.Context, i int, m *TestMiner) {
+			defer wg.Done()
+			m.mirDB = NewTestDB()
+			m.checkpointPeriod = checkpoint
```

Comment:
You removed `checkpointPeriod` from the validator, right? Why is this still here?

Reply:
Correct. Removed.